src/java.base/share/classes/jdk/internal/foreign/HeapMemorySegmentImpl.java

```diff
@@ -59,9 +61,19 @@ public abstract class HeapMemorySegmentImpl extends AbstractMemorySegmentImpl {
     final long offset;
     final Object base;
 
+    @Override
+    public Optional<Object> array() {
+        return Optional.of(base);
```

Comment:
I wonder if we should use here `ofNullable`

Reply:
I would like to recall this comment. Even if it would look better from language perspective, having not specialized `Optional` can make `Unsafe` operations slower, as the VM can make issues with detecting if it's on heap or off-heap access.